### ISSUE TITLE
WMR headsets finally working with latest Steam VR Mixed Reality Patch !!

### ISSUE THREAD

**Issue by SilentChill** [LukeRoss00/gta5-real-mod#285]
Not an issue but worth saying that its finally not double vision !!!

Amazing on a Reverb G2 now !

**Comment by Dark-CSS** [comment]
Does the 3d (alternate eye rendering) works fine with the G2? Rift S user here.

**Comment by ODEH67** [comment]
@SilentChill can u explain more what do u mean that WMR headset finally works with the latest patch? because im a Reverb G1 user and my G2 comes tomorrow, the mod is working already for months now on the WMR headsets :\ ....

https://github.com/LukeRoss00/gta5-real-mod/issues/78#issuecomment-648764343 .

since then the mod worked better with WMR

**Comment by SilentChill** [comment]
@ODEH67 when running through steam vr you get double vision unless using openxr through WMR , now you can enable OXR in steam and there are no issues.   

Reason I want to use steam vr is is for motion compensation  as I have a 6DOF rig

**Comment by ODEH67** [comment]
@SilentChill So u need to run openxr in wmr and the OXR at the same time in order to make it work or is the OXR in steam is enough?

**Comment by emozio666** [comment]
that is, i can now play normally with Pimax?

**Comment by softrare** [comment]
> 
> 
> @ODEH67 when running through steam vr you get double vision unless using openxr through WMR , now you can enable OXR in steam and there are no issues.
> 
> Reason I want to use steam vr is is for motion compensation as I have a 6DOF rig

I have tried OXR through WMR and SteamVR, I get the double vision thing through both, that is it is reasonaly clear as long as I don't move my head. Is this really fixed? How can I take advantage of that fix? HP Reverb G2 here.

**Comment by LukeRoss00** [comment]
@softrare Make sure that you selected the OpenXR backend of the mod by changing the last line of `RealVR.ini` to `VRAPI = 3`. When the OpenXR runtime for WMR is active, SteamVR should not even be loaded.

**Comment by softrare** [comment]
Sorry, couldn't answer I was more or less only in VR up until now from the moment I tried your solution :D It works and is absolutely fantastic. Where can I donate to you?

**Comment by LukeRoss00** [comment]
I'm not taking donations for the moment, but thanks for the appreciation!!

**Comment by virtualmisterl** [comment]
> @ODEH67 when running through steam vr you get double vision unless using openxr through WMR , now you can enable OXR in steam and there are no issues.
> 
> Reason I want to use steam vr is is for motion compensation as I have a 6DOF rig

Hey @SilentChill 
Was wondering if you can help me out with a little more info.
I also have a motion rig & reverb g2 but I can't get a jitter free experience with steam vr running.

If I set the RealVR.ini to VRAPI = 3 then I get a perfect jitter free experience without using steam vr.
But the minute I use steam vr its back to a jittery mess when I move my head.

How fo I enable oxr in steam?

I'm on the beta, my openxr runtime is set to steamvr under developer tab & can't be changed.
I have wmr for steamvr installed on steam, I have openxr developer tools installed in windows.
I've tried it set with vrapi =0 & vrapi=3
Basically if I load it without steam its fine, if I change settings to load with steam then its jittery.

If you could go a bit more in detail about exactly how to set this up I would be grateful as I'm new to the wmr software & I find it all a bit of a buggy mess

**Comment by softrare** [comment]
@LukeRoss00 I have a weird issue where suddenly the image of the game rotates with my head movement thus being completely 2D within my G2. I don't think I changed anything and where I did I changed it back. Have you ever encountered this issue?

**Comment by LukeRoss00** [comment]
I can't say that I have, although I'm not sure that I understand exactly what you mean. Are you using other mods? You might try reinstalling the VR mod to be on the safe side, and if that doesn't work please open a new issue on the tracker.

**Comment by LukeRoss00** [comment]
@SilentChill @virtualmisterl The mod started working correctly with SteamVR on Jan 23, when Microsoft enabled support to alternate eye rendering with[Windows Mixed Reality for SteamVR [beta 1.2.485](https://store.steampowered.com/news/app/719950/view/3023572386954504207). Unfortunately, they kind of botched the update by throwing in a new bug that broke Google Earth VR, so people started screaming bloody murder and Microsoft rolled back the fix.

If you want to play with SteamVR, the only thing you can do at the moment (just verified and it works, although it's a nuisance) is rolling back from the current version 1.2.536 to the 1.2.485 beta, which unfortunately must be done via a Steam depot download tool like https://github.com/SteamRE/DepotDownloader. Alternatively you can try lobbying the Microsoft guys so that they provide an official way of reverting to 1.2.485 (basically they should create a new channel that you could then select on Steam).

**Comment by virtualmisterl** [comment]
> @SilentChill @virtualmisterl The mod started working correctly with SteamVR on Jan 23, when Microsoft enabled support to alternate eye rendering with Windows Mixed Reality for SteamVR [version 1.2.485](https://store.steampowered.com/news/app/719950/view/3023572386954504207). Unfortunately, they kind of botched the update by throwing in a new bug that broke Google Earth VR, so people started screaming bloody murder and Microsoft rolled back the fix.
> 
> If you want to play with SteamVR, the only thing you can do at the moment (just verified and it works, although it's a nuisance) is rolling back from the current version 1.2.536 to the old 1.2.485, which unfortunately must be done via a Steam depot download tool like https://github.com/SteamRE/DepotDownloader. Alternatively you can try lobbying the Microsoft guys so that they provide an official way of reverting to 1.2.485 (basically they should create a new channel that you could then select on Steam).

Thanks but it turns out 1.2.485 has other problems for my use.
I actually had it running with that version after speaking to @SilentChill & moving to the wmr for steamvr beta branch.
but I can't use motion compensation for my motion simulator in steam overlay with as it is unuseable on 1.2 485.
Basically when ovrmc tracker is enabled rather than cancelling the rig movement, the game turns into a kind of 2d rectangle window & as I turn, it turns my view out of the rectangle window into pitch black nothingness.
Only disabling my tracker in ovrmc brings it back.
As soon as I disabled the wmr for steam vr beta, that issue was gone but the jitter returned.

I don't really know the problem exactly but it seems maybe overlays have been broken & can't be used at the same time as the alternate eye rendering pose stuff they implemented.
Like they arent integrated together properly & interfere with each other.
Like the 2

**Comment by LukeRoss00** [comment]
Unfortunately, based on what you just posted and what [/u/TymAtMSFT](https://www.reddit.com/user/TymAtMSFT/) wrote [here](https://www.reddit.com/r/WindowsMR/comments/hezaeu/will_alternate_eye_rendering_needed_for_my_gta_vr/fvw99zw), I'd say that there is definitely a link between the two problems and it could be a long time before they sort it out. With OpenVR becoming legacy and OpenXR getting more popular, it's quite possible that they'll never get around to it.
